cli/lsp/language_server.rs

```diff
@@ -2157,11 +2283,30 @@ impl Inner {
         specifier,
         line_index.offset_tsc(params.text_document_position_params.position)?,
         asset_or_doc.scope().cloned(),
+        token,
       )
-      .await?;
+      .await
+      .map_err(|err| {
```

Comment:
Why only `map_err`? IIUC this will keep iterating if items return ok

Reply:
These are all `Result::map_err()`, not sure what you mean. What iteration?

Reply:
Nvm, I think I confused it with some other place that did `.into_iter().map(...)` and only then did `.map_err()` and checked for token

Reply:
Yeah in other places I did `.collect::<Result<Vec<_>, _>>()`, these will stop on the first error though.